### ISSUE TITLE
Bug: `_undersample_data` method ignores `experiment_num` parameter, causing identical sampling results across multiple experiments

### ISSUE THREAD

**Issue by 1602353775** [embeddings-benchmark/mteb#3896]
### Description

In classification task evaluation, the `_undersample_data` method receives the `experiment_num` parameter but does not use it when initializing the random state. This causes all experiments (`n_experiments`) to use the same random seed, resulting in identical or highly similar sampling results, which prevents true Bootstrap sampling.

### Current Behavior

In the `_evaluate_subset` method of `mteb/abstasks/classification.py`, `_undersample_data` is called in a loop for multiple experiments:
```py
for i in range(self.n_experiments):
    train_dataset, idxs = self._undersample_data(
        train_split,
        i,  # experiment_num is passed
        idxs,
    )
```
However, in the _undersample_data method (line 294), the random state only uses self.seed:
```
rng_state = np.random.RandomState(self.seed)  # experiment_num is not used rng_state.shuffle(idxs)
```
This causes:
All experiments use the same random seed
The shuffled index results are identical each time
Sampling results are identical or highly similar
Cannot reflect the randomness and statistical significance of Bootstrap sampling
Expected Behavior
According to the method's comment "Bootstrap self.samples_per_label samples per label for each split", each experiment should use a different random seed to achieve true Bootstrap sampling, which would:
Sample different training sets for each experiment
Enable calculation of mean and standard deviation across multiple experiments
Better evaluate model stability and generalization ability
Steps to Reproduce
Run any classification task with n_experiments > 1 (e.g., n_experiments=10)
Add logging in the _undersample_data method to print sampled_idxs for each experiment
Observe that sampled_idxs are identical across all experiments

### To reproduce
```py
rng_state = np.random.RandomState(self.seed)
if experiment_num > 0:
    rng_state = np.random.RandomState(self.seed + experiment_num)
```
Impact
Affected task types: All classification tasks inheriting from AbsTaskClassification
Affected functionality: Statistical significance of multiple experiments (mean, standard deviation, etc.)
Severity: Medium - Does not affect single experiment results, but impacts the statistical validity of multiple experiments

### Are you interested to contribute a fix for this bug?

No

**Comment by Samoed** [comment]
Thank you for investigation, but indexes will be different for each experiment. On first step `idxs` will be just `list(range(len(dataset)))`, but on each next experiment it will receive already shuffled `idxs` and they're won't be the same. 

>  Observe that sampled_idxs are identical across all experiments

If I print `sampled_idxs[:10], idxs[:10]` after each experiment on `Banking77Classification`, I will receive 

```
Experiment 0 sampled idxs [6883, 5836, 8601, 2545, 8697, 5573, 576, 6832, 7111, 439] idxs [6883, 5836, 8601, 2545, 8697, 5573, 576, 6832, 7111, 439]
Experiment 1 sampled idxs [1774, 4068, 3356, 8599, 7657, 6646, 2122, 4144, 4088, 2191] idxs [1774, 4068, 3356, 8599, 7657, 6646, 2122, 4144, 4088, 2191]
Experiment 2 sampled idxs [3014, 5020, 3879, 6540, 8192, 6772, 5868, 2999, 373, 6449] idxs [3014, 5020, 3879, 6540, 8192, 6772, 5868, 2999, 373, 6449]
Experiment 3 sampled idxs [7365, 9504, 5584, 2241, 3083, 6538, 1293, 3065, 3002, 6352] idxs [7365, 9504, 5584, 2241, 3083, 6538, 1293, 3065, 3002, 6352]
Experiment 4 sampled idxs [5055, 9825, 6360, 4961, 4038, 3606, 6942, 1680, 9072, 5923] idxs [5055, 9825, 6360, 4961, 4038, 3606, 6942, 1680, 9072, 5923]
Experiment 5 sampled idxs [3779, 698, 2668, 2431, 2887, 549, 8545, 2098, 1865, 1752] idxs [3779, 698, 2668, 2431, 2887, 549, 8545, 2098, 1865, 1752]
Experiment 6 sampled idxs [2263, 5397, 7551, 2164, 3057, 2908, 9914, 473, 9046, 8732] idxs [2263, 5397, 7551, 2164, 3057, 2908, 9914, 473, 9046, 8732]
Experiment 7 sampled idxs [9602, 8727, 9187, 8832, 1420, 2663, 5463, 1224, 827, 2033] idxs [9602, 8727, 9187, 8832, 1420, 2663, 5463, 1224, 827, 2033]
Experiment 8 sampled idxs [1081, 8623, 1733, 2343, 3994, 3782, 1788, 743, 4541, 7838] idxs [1081, 8623, 1733, 2343, 3994, 3782, 1788, 743, 4541, 7838]
Experiment 9 sampled idxs [8084, 7059, 1803, 5354, 5391, 8500, 3749, 47, 6661, 772] idxs [8084, 7059, 1803, 5354, 5391, 8500, 3749, 47, 6661, 772]
```

And each experiment would have different idxs
